### DISCUSSION TITLE
Integrations with Zod

### DISCUSSION THREAD

**Discussion by achaconm2001** [ivan-dalmet/formiz#73] [Ideas]
Hi dude, I have being using formiz a lot in the last few months and I really like it. In my current job I have to integrated with an existing validation set made with zod. I was searching but I could not find any example or even idea. I don't know if you have some example of how to made that. :)

**Comment by ivan-dalmet** [comment]
Hi @Achaconm2001 
I didn't have time to try it but it should works with the `rule` property of `validations`.

Something like that should work I think (not tested code)

```jsx
<MyField
  name="field-name"
  validations={[
    { 
      rule: (value) => z.string().min(5).parse(value), 
      message: 'String should be longer that 5 characters', 
    }
  ]}
/>
```

You can extract this in a custom rule (also not tested code 😉 ):
```jsx
const zodRule = (zod) => (value) => zod.parse(value);

<MyField
  name="field-name"
  validations={[
    { 
      rule: zodRule(z.string().min(5)), 
      message: 'String should be longer that 5 characters', 
    }
  ]}
/>
```

or also 
```jsx
const zodRule = (zod) => (value) => zod.parse(value);
const ruleStringMin = (min) => zodRule(z.string().min(min));

<MyField
  name="field-name"
  validations={[
    { 
      rule: ruleStringMin(5), 
      message: 'String should be longer that 5 characters', 
    }
  ]}
/>
```
